src/config.c

```diff
@@ -462,8 +485,12 @@ int parseCommandLine(int argc,char **argv,ParodusCfg * cfg)
          break;
         
         case 's':
-          parStrncpy(cfg->hw_serial_number,optarg,sizeof(cfg->hw_serial_number));
-          ParodusInfo("hw_serial_number is %s\n",cfg->hw_serial_number);
+	if(parse_serial_num(cfg->hw_serial_number, optarg) == 0) {
+            ParodusInfo ("hw_serial-number is %s\n",cfg->hw_serial_number);
+					} else {
+						ParodusError ("Bad serial-number %s\n", optarg);
+						return -1;
```

Comment:
I think return is not required as serial number is not a mandatory argument. we should just log an error and proceed with other arguments

Reply:
For junk value lets log error and set to default unknown as its not mandatory